tools/manifest/sourcefile.py

```diff
@@ -191,15 +216,16 @@ def __init__(self, tests_root, rel_path, url_base, hash=None, contents=None):
         self.name, self.ext = os.path.splitext(self.filename)
 
         self.type_flag = None
-        if "-" in self.name:
-            self.type_flag = self.name.rsplit("-", 1)[1].split(".")[0]
+        if b"-" in self.name:
+            self.type_flag = self.name.rsplit(b"-", 1)[1].split(b".")[0]
 
-        self.meta_flags = self.name.split(".")[1:]
+        self.meta_flags = self.name.split(b".")[1:]
```

Comment:
At least this is problematic for the lint:
```py
def check_parsed(repo_root, path, f):
    # type: (str, str, IO[bytes]) -> List[rules.Error]
    source_file = SourceFile(repo_root, path, "/", contents=f.read())
```
where the first two arguments are treated as `bytes` here.

Reply:
This still ain't great, but it's good enough for MyPy to be happy in Py2 mode node.